### PR TITLE
miniscript/types: check that timelock values aren't too large

### PR DESCRIPTION
**Pull request by darosior**
Wonder if we should check that the timelock value is minimal too, what do you think?

### CODE REVIEW

File: src/miniscript/types/extra_props.rs [hunk 1]
```diff
@@ -851,20 +853,19 @@ impl Property for ExtData {
                 Ok(Self::from_multi(k, pks.len()))
             }
             Terminal::After(t) => {
-                // FIXME check if t > 2^31 - 1
-                if t == 0 {
+                if t == 0 || (t & SEQUENCE_LOCKTIME_DISABLE_FLAG) == 1 {
```
Comment (sanket1729) [inline src/miniscript/types/extra_props.rs:856]: I don't think that there are any rules applicable for after fragment. The disable flag is only for CSV.

Comment (darosior) [reply src/miniscript/types/extra_props.rs]: Right, i was confused by the FIXME comment and didn't double check.. FIxed, thanks!

Comment (darosior) [reply src/miniscript/types/extra_props.rs:859]: Reverted post discussion on IRC and added a rationale comment so we don't forget about it :)

**Review by sanket1729 — APPROVED** [review]
utACK a3dd9fe840c8cdae84e9d586cd9a9d6c195f0092 . Thanks for doing this.

**Review by apoelstra — APPROVED** [review]
ACK a3dd9fe840c8cdae84e9d586cd9a9d6c195f0092

**Pull request merged**